### PR TITLE
Add UEFI Secure Boot with unsigned kernel case (New)

### PR DESCRIPTION
**Pull request by baconYao**
## Description

Currently, for secure boot coverage, Checkbox only has [miscellanea/secure_boot_mode](https://github.com/canonical/checkbox/blob/v4.1.0/providers/base/units/miscellanea/jobs.pxu#L133) to check secure boot is enabled from System level.

However, based on the responsibility of publishing image, Canonical is responsible for the signed shim, grub and kernel. Therefore, I add a manual case to replace **_signed kernel_** with **_unsigned kernel_** to prove the function of UEFI Secure Boot is working as expected.

This is raised from Boashan Project but the testing method is generic. So Canonical QA can follow the step to verify UEFI Secure Boot feature if a project supports UEFI firmware on **_Classic_** image.

## Resolved issues

- https://warthogs.atlassian.net/browse/OEMQA-4820

## Documentation

- [Baoshan - Secure Boot Enablement Guide](https://docs.google.com/document/d/1s4cbuRELmthRa2Z_yJSLMzxaqQPLDoHM3lra0_0avII/edit)
- https://forums.debian.net/viewtopic.php?p=789903&sid=8de6b14ace62a6f1fbe9103aa35c0346#p789903
- https://sources.debian.org/src/grub2/2.06-13+deb12u1/grub-core/kern/efi/sb.c/#L182
- https://launchpad.net/~canonical-kernel-team/+archive/ubuntu/ppa/+packages?field.name_filter=&field.status_filter=published&field.series_filter=jammy

## Tests
Screenshot from 2024-08-30 10-04-30

Screenshot from 2024-08-30 10-04-50

Screenshot from 2024-08-29 15-06-03

### CODE REVIEW

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/test-plan-ce-oem.pxu [hunk 1]
```diff
@@ -82,6 +83,7 @@ nested_part:
     com.canonical.certification::rtc-automated
     com.canonical.certification::led-indicator-auto
     before-suspend-ce-oem-spi-automated
+    ce-oem-secure-boot-automated
```
Comment (hanhsuan) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/test-plan-ce-oem.pxu:86]: It should be better to add some description for this empty test plan

Comment (baconYao) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/test-plan-ce-oem.pxu:86]: I will not include `ce-oem-secure-boot-automated` test plan since it's empty. Thanks

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/secure-boot/jobs.pxu [hunk 2]
```diff
@@ -0,0 +1,28 @@
+plugin: manual
+category_id: secure-boot
```
Comment (stanley31huang) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/secure-boot/jobs.pxu:2]: we could use the security category in base provider. It seems the same testing purpose.
https://github.com/canonical/checkbox/blob/main/providers/base/units/security/category.pxu

Comment (baconYao) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/secure-boot/jobs.pxu:2]: Roger that!

**Comment by baconYao** [comment]
@hanhsuan @stanley31huang I revert the commit to fix the issue of 80 length limitation. We should file another PR to fix it.

**Review by stanley31huang — APPROVED** [review]
LGTM
